<img SRC="img01.png"/>
<center><h1 Style="color:#DC143C">Analisis de Datos</h1></center>
<center><h5>Universidad Catro Carazo</h5></center>
</br>
<center><h6>2C - 2022</h6></center>
</br>
</br>
<p Style="text-aling: justify">El carro de la presente imagen, es un carro antiguo de los años 70 y 80 s. </p>
<a  href="https://archive.ics.uci.edu/ml/index.php">Fuente de datos para el analisis</a>


<h2 Style="color:#DC143C">1.Carga y estandarizacion de los set de datos</h2>
<p> Que se hace en esta etapa?</p>

<o1>
    
    <1i>Importar las librerias necesarias</1i>
    <1i>Importar el o los set de datos </1i>
    <1i>Revisar los encabezados de los dataframes</1i>
    
    
</o1> 
<h3 style="color:#DC143C">1.1. Importar las librerias </h3>
    

In [1]:
import pandas as pd                 #se usa para el manejo de datafrom 
import numpy as np                  #se usa para el manejo de estructura de datos 
import matplotlib.pyplot as ptl     #se usa para el manejo de graficos 

<h3 Style ="color:#DC143C">1.2. Importar el set de datos </h3>

In [2]:
cols = ["symboling", "normalized", "make", "fuel", "aspiration", "doors",
"style", "wheels", "engine", "base", "length", "width", "heingt",
"curb", "engineType", "Cylinders", "engine-size", "system", "bore", "stroke",
"compression", "horsepower", "peak", "City", "Higway", "price"]

df_1er_autos= pd.read_csv("autos.csv", names=cols)
df_2do_autos = pd.read_csv("Autos1.csv", names=cols)

In [3]:
df_1er_autos
df_2do_autos

symboling normalized        make    fuel aspiration doors        style  \
0           2        134      toyota     gas        std   two      hardtop   
1           2        134      toyota     gas        std   two    hatchback   
2           2        134      toyota     gas        std   two      hardtop   
3           2        134      toyota     gas        std   two    hatchback   
4           2        134      toyota     gas        std   two  convertible   
5          -1         65      toyota     gas        std  four        sedan   
6          -1         65      toyota  diesel      turbo  four        sedan   
7          -1         65      toyota     gas        std  four    hatchback   
8          -1         65      toyota     gas        std  four        sedan   
9          -1         65      toyota     gas        std  four    hatchback   
10          3        197      toyota     gas        std   two    hatchback   
11          3        197      toyota     gas        std   two    hatchback   
12         -1         90      toyota     gas        std  four        sedan   
13         -1          ?      toyota     gas        std  four        wagon   
14          2        122  volkswagen  diesel        std   two        sedan   
15          2        122  volkswagen     gas        std   two        sedan   
16          2         94  volkswagen  diesel        std  four        sedan   
17          2         94  volkswagen     gas        std  four        sedan   
18          2         94  volkswagen     gas        std  four        sedan   
19          2         94  volkswagen  diesel      turbo  four        sedan   
20          2         94  volkswagen     gas        std  four        sedan   
21          3          ?  volkswagen     gas        std   two  convertible   
22          3        256  volkswagen     gas        std   two    hatchback   
23          0          ?  volkswagen     gas        std  four        sedan   
24          0          ?  volkswagen  diesel      turbo  four        sedan   
25          0          ?  volkswagen     gas        std  four        wagon   
26         -2        103       volvo     gas        std  four        sedan   
27         -1         74       volvo     gas        std  four        wagon   
28         -2        103       volvo     gas        std  four        sedan   

   wheels engine   base  ...  engine-size  system  bore  stroke compression  \
0     rwd  front   98.4  ...          146    mpfi  3.62    3.50         9.3   
1     rwd  front   98.4  ...          146    mpfi  3.62    3.50         9.3   
2     rwd  front   98.4  ...          146    mpfi  3.62    3.50         9.3   
3     rwd  front   98.4  ...          146    mpfi  3.62    3.50         9.3   
4     rwd  front   98.4  ...          146    mpfi  3.62    3.50         9.3   
5     fwd  front  102.4  ...          122    mpfi  3.31    3.54         8.7   
6     fwd  front  102.4  ...          110     idi  3.27    3.35        22.5   
7     fwd  front  102.4  ...          122    mpfi  3.31    3.54         8.7   
8     fwd  front  102.4  ...          122    mpfi  3.31    3.54         8.7   
9     fwd  front  102.4  ...          122    mpfi  3.31    3.54         8.7   
10    rwd  front  102.9  ...          171    mpfi  3.27    3.35         9.3   
11    rwd  front  102.9  ...          171    mpfi  3.27    3.35         9.3   
12    rwd  front  104.5  ...          171    mpfi  3.27    3.35         9.2   
13    rwd  front  104.5  ...          161    mpfi  3.27    3.35         9.2   
14    fwd  front   97.3  ...           97     idi  3.01    3.40        23.0   
15    fwd  front   97.3  ...          109    mpfi  3.19    3.40         9.0   
16    fwd  front   97.3  ...           97     idi  3.01    3.40        23.0   
17    fwd  front   97.3  ...          109    mpfi  3.19    3.40         9.0   
18    fwd  front   97.3  ...          109    mpfi  3.19    3.40         9.0   
19    fwd  front   97.3  ...           97     idi  3.01    3.40        23.0   
20  

In [4]:
df_1er_autos.shape #esto me permite conocer el tamaño del DataFrame

(197, 26)

In [5]:
df_2do_autos.shape 

(29, 26)

In [6]:
df_1er_autos.head(10)  #10 significa la cantidad de filas que quiero analizar, si quiero no le pongo cantidad por lo tanto quito parentesis o si quiero pongo la cantidad 

symboling normalized         make fuel aspiration doors        style  \
0          3          ?  alfa-romero  gas        std   two  convertible   
1          3          ?  alfa-romero  gas        std   two  convertible   
2          1          ?  alfa-romero  gas        std   two    hatchback   
3          2        164         audi  gas        std  four        sedan   
4          2        164         audi  gas        std  four        sedan   
5          2          ?         audi  gas        std   two        sedan   
6          1        158         audi  gas        std  four        sedan   
7          1          ?         audi  gas        std  four        wagon   
8          1        158         audi  gas      turbo  four        sedan   
9          0          ?         audi  gas      turbo   two    hatchback   

  wheels engine   base  ...  engine-size  system  bore  stroke compression  \
0    rwd  front   88.6  ...          130    mpfi  3.47    2.68         9.0   
1    rwd  front   88.6  ...          130    mpfi  3.47    2.68         9.0   
2    rwd  front   94.5  ...          152    mpfi  2.68    3.47         9.0   
3    fwd  front   99.8  ...          109    mpfi  3.19    3.40        10.0   
4    4wd  front   99.4  ...          136    mpfi  3.19    3.40         8.0   
5    fwd  front   99.8  ...          136    mpfi  3.19    3.40         8.5   
6    fwd  front  105.8  ...          136    mpfi  3.19    3.40         8.5   
7    fwd  front  105.8  ...          136    mpfi  3.19    3.40         8.5   
8    fwd  front  105.8  ...          131    mpfi  3.13    3.40         8.3   
9    4wd  front   99.5  ...          131    mpfi  3.13    3.40         7.0   

  horsepower  peak City Higway  price  
0        111  5000   21     27  13495  
1        111  5000   21     27  16500  
2        154  5000   19     26  16500  
3        102  5500   24     30  13950  
4        115  5500   18     22  17450  
5        110  5500   19     25  15250  
6        110  5500   19     25  17710  
7        110  5500   19     25  18920  
8        140  5500   17     20  23875  
9        160  5500   16     22      ?  

[10 rows x 26 columns]

In [7]:
df_1er_autos.tail(10) #tail es cola osea que va a seleccionar las ultimas lineas con head que es cabeza selecciona las primeras lineas 

symboling normalized        make    fuel aspiration doors        style  \
187          2         94  volkswagen  diesel      turbo  four        sedan   
188          2         94  volkswagen     gas        std  four        sedan   
189          3          ?  volkswagen     gas        std   two  convertible   
190          3        256  volkswagen     gas        std   two    hatchback   
191          0          ?  volkswagen     gas        std  four        sedan   
192          0          ?  volkswagen  diesel      turbo  four        sedan   
193          0          ?  volkswagen     gas        std  four        wagon   
194         -2        103       volvo     gas        std  four        sedan   
195         -1         74       volvo     gas        std  four        wagon   
196         -2        103       volvo     gas        std  four        sedan   

    wheels engine   base  ...  engine-size  system  bore  stroke compression  \
187    fwd  front   97.3  ...           97     idi  3.01    3.40        23.0   
188    fwd  front   97.3  ...          109    mpfi  3.19    3.40        10.0   
189    fwd  front   94.5  ...          109    mpfi  3.19    3.40         8.5   
190    fwd  front   94.5  ...          109    mpfi  3.19    3.40         8.5   
191    fwd  front  100.4  ...          136    mpfi  3.19    3.40         8.5   
192    fwd  front  100.4  ...           97     idi  3.01    3.40        23.0   
193    fwd  front  100.4  ...          109    mpfi  3.19    3.40         9.0   
194    rwd  front  104.3  ...          141    mpfi  3.78    3.15         9.5   
195    rwd  front  104.3  ...          141    mpfi  3.78    3.15         9.5   
196    rwd  front  104.3  ...          141    mpfi  3.78    3.15         9.5   

    horsepower  peak City Higway  price  
187         68  4500   37     42   9495  
188        100  5500   26     32   9995  
189         90  5500   24     29  11595  
190         90  5500   24     29   9980  
191        110  5500   19     24  13295  
192         68  4500   33     38  13845  
193         88  5500   25     31  12290  
194        114  5400   23     28  12940  
195        114  5400   23     28  13415  
196        114  5400   24     28  15985  

[10 rows x 26 columns]

In [8]:
df_1er_autos.describe() # esta funcion describe los valores estadisticamente 

symboling        base      length       width      heingt  \
count  197.000000  197.000000  197.000000  197.000000  197.000000   
mean     0.913706   98.409645  173.450254   65.812690   53.628934   
std      1.202746    5.866965   12.213912    2.128306    2.438891   
min     -2.000000   86.600000  141.100000   60.300000   47.800000   
25%      0.000000   94.500000  166.300000   64.000000   52.000000   
50%      1.000000   96.500000  172.600000   65.400000   53.900000   
75%      2.000000  100.400000  178.500000   66.500000   55.500000   
max      3.000000  120.900000  208.100000   72.300000   59.800000   

              curb  engine-size  compression        City      Higway  
count   197.000000   197.000000   197.000000  197.000000  197.000000  
mean   2534.796954   126.263959    10.128020   25.416244   30.984772  
std     520.401159    42.282594     3.934467    6.566717    6.908644  
min    1488.000000    61.000000     7.000000   13.000000   16.000000  
25%    2140.000000    97.000000     8.600000   19.000000   25.000000  
50%    2405.000000   110.000000     9.000000   25.000000   30.000000  
75%    2844.000000   141.000000     9.400000   30.000000   36.000000  
max    4066.000000   326.000000    23.000000   49.000000   54.000000

<h3 Style ="color:#DC143C">1.3. Revisar encabezados de los set de datos </h3>

In [9]:
df_1er_autos.info () #informacion especifica de una columna 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 197 entries, 0 to 196
Data columns (total 26 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   symboling    197 non-null    int64  
 1   normalized   197 non-null    object 
 2   make         197 non-null    object 
 3   fuel         197 non-null    object 
 4   aspiration   197 non-null    object 
 5   doors        197 non-null    object 
 6   style        197 non-null    object 
 7   wheels       197 non-null    object 
 8   engine       197 non-null    object 
 9   base         197 non-null    float64
 10  length       197 non-null    float64
 11  width        197 non-null    float64
 12  heingt       197 non-null    float64
 13  curb         197 non-null    int64  
 14  engineType   197 non-null    object 
 15  Cylinders    197 non-null    object 
 16  engine-size  197 non-null    int64  
 17  system       197 non-null    object 
 18  bore         197 non-null    object 
 19  stroke  

In [10]:
df_1er_autos.columns #solo para ver las columnas y no todo el archivo 

Index(['symboling', 'normalized', 'make', 'fuel', 'aspiration', 'doors',
       'style', 'wheels', 'engine', 'base', 'length', 'width', 'heingt',
       'curb', 'engineType', 'Cylinders', 'engine-size', 'system', 'bore',
       'stroke', 'compression', 'horsepower', 'peak', 'City', 'Higway',
       'price'],
      dtype='object')

In [11]:
df_2do_autos.columns

Index(['symboling', 'normalized', 'make', 'fuel', 'aspiration', 'doors',
       'style', 'wheels', 'engine', 'base', 'length', 'width', 'heingt',
       'curb', 'engineType', 'Cylinders', 'engine-size', 'system', 'bore',
       'stroke', 'compression', 'horsepower', 'peak', 'City', 'Higway',
       'price'],
      dtype='object')

#INICIA LA SEGUNDA SEMANA 

<h3 Style="color:#8B008B">1.4 Unir archivos en un solo DataFrame</h3> 

In [13]:
unionArchivos = [df_1er_autos, df_2do_autos] #lista de DataFrame
dfAutos = pd.concat(unionArchivos)
dfAutos.shape #en el parentesis de abajo el primer dato es la cantidad de registros y el segundo cantidad de archivos 

(226, 26)

In [14]:
dfAutos.head(3)

symboling normalized         make fuel aspiration doors        style  \
0          3          ?  alfa-romero  gas        std   two  convertible   
1          3          ?  alfa-romero  gas        std   two  convertible   
2          1          ?  alfa-romero  gas        std   two    hatchback   

  wheels engine  base  ...  engine-size  system  bore  stroke compression  \
0    rwd  front  88.6  ...          130    mpfi  3.47    2.68         9.0   
1    rwd  front  88.6  ...          130    mpfi  3.47    2.68         9.0   
2    rwd  front  94.5  ...          152    mpfi  2.68    3.47         9.0   

  horsepower  peak City Higway  price  
0        111  5000   21     27  13495  
1        111  5000   21     27  16500  
2        154  5000   19     26  16500  

[3 rows x 26 columns]

# A CONTINUACION FORMAS PARA SEPARAR FILAS Y COLUMNAS 

In [17]:
dftem=dfAutos.iloc[:,0:2] #el primer espacio : antes de la coma es = a la cantidad de filas, el segundo campo despues de la coma significa la cantidd de columnas que deseo seleccionar 

In [18]:
dftem.head(6) #se usa para cortar columnas , depende la version de python se usa loc o iloc para las mas recientes, este proceso se usa para separar columnas 

symboling normalized
0          3          ?
1          3          ?
2          1          ?
3          2        164
4          2        164
5          2          ?

In [20]:
dftem=dfAutos.iloc[[0,3,6,24],[0,5,6]]

In [21]:
dftem

symboling doors        style
0           3   two  convertible
3           2  four        sedan
6           1  four        sedan
24          1  four    hatchback

<h2 Style="color:#DAA520">2. Manejo de valores faltantes o perdidos </h2>
<p> Que se hace en esta etapa?</p>

<ol>
    <li>Como se muestran los valores faltantes (null, espacio en blanco o vacio, o un signo de ?) y estandarizarlos (NaN(nom and number))</li>
    <li>Contar la cantidad de valores de NaN, faltantes por variables (columna) </li>
    <li>Analisis de imputacion(el promedio / se usa tambien procesos de MACHINE LEARNING) para el caso de los numericos </li>
    <li>Analisis de imputacion(con el valor mas frecuente de la columna) para el caso de los caracteres </li>
    <li> Eliminacion de filas y columnas con valores faltantes  </li> 
</ol> 

<h3 style="#DAA520">2.1. Identificar como se muestran los valores faltantes y estandarizarlos </h3>
    

In [43]:
dfAutos.head(15)

symboling normalized         make fuel aspiration doors        style  \
0           3        NaN  alfa-romero  gas        std   two  convertible   
1           3        NaN  alfa-romero  gas        std   two  convertible   
2           1        NaN  alfa-romero  gas        std   two    hatchback   
3           2        164         audi  gas        std  four        sedan   
4           2        164         audi  gas        std  four        sedan   
5           2        NaN         audi  gas        std   two        sedan   
6           1        158         audi  gas        std  four        sedan   
7           1        NaN         audi  gas        std  four        wagon   
8           1        158         audi  gas      turbo  four        sedan   
9           0        NaN         audi  gas      turbo   two    hatchback   
10          2        192          bmw  gas        std   two        sedan   
11          0        192          bmw  gas        std  four        sedan   
12          0        188          bmw  gas        std   two        sedan   
13          0        188          bmw  gas        std  four        sedan   
14          1        NaN          bmw  gas        std  four        sedan   

   wheels engine   base  ...  engine-size  system  bore  stroke compression  \
0     rwd  front   88.6  ...          130    mpfi  3.47    2.68         9.0   
1     rwd  front   88.6  ...          130    mpfi  3.47    2.68         9.0   
2     rwd  front   94.5  ...          152    mpfi  2.68    3.47         9.0   
3     fwd  front   99.8  ...          109    mpfi  3.19    3.40        10.0   
4     4wd  front   99.4  ...          136    mpfi  3.19    3.40         8.0   
5     fwd  front   99.8  ...          136    mpfi  3.19    3.40         8.5   
6     fwd  front  105.8  ...          136    mpfi  3.19    3.40         8.5   
7     fwd  front  105.8  ...          136    mpfi  3.19    3.40         8.5   
8     fwd  front  105.8  ...          131    mpfi  3.13    3.40         8.3   
9     4wd  front   99.5  ...          131    mpfi  3.13    3.40         7.0   
10    rwd  front  101.2  ...          108    mpfi  3.50    2.80         8.8   
11    rwd  front  101.2  ...          108    mpfi  3.50    2.80         8.8   
12    rwd  front  101.2  ...          164    mpfi  3.31    3.19         9.0   
13    rwd  front  101.2  ...          164    mpfi  3.31    3.19         9.0   
14    rwd  front  103.5  ...          164    mpfi  3.31    3.19         9.0   

   horsepower  peak City Higway  price  
0         111  5000   21     27  13495  
1         111  5000   21     27  16500  
2         154  5000   19     26  16500  
3         102  5500   24     30  13950  
4         115  5500   18     22  17450  
5         110  5500   19     25  15250  
6         110  5500   19     25  17710  
7         110  5500   19     25  18920  
8         140  5500   17     20  23875  
9         160  5500   16     22    NaN  
10        101  5800   23     29  16430  
11        101  5800   23     29  16925  
12        121  4250   21     28  20970  
13        121  4250   21     28  21105  
14        121  4250   20     25  24565  

[15 rows x 26 columns]

In [46]:
dfAutos.replace("?", np.nan, inplace = True)
dfAutos.head(10)

symboling normalized         make fuel aspiration doors        style  \
0          3        NaN  alfa-romero  gas        std   two  convertible   
1          3        NaN  alfa-romero  gas        std   two  convertible   
2          1        NaN  alfa-romero  gas        std   two    hatchback   
3          2        164         audi  gas        std  four        sedan   
4          2        164         audi  gas        std  four        sedan   
5          2        NaN         audi  gas        std   two        sedan   
6          1        158         audi  gas        std  four        sedan   
7          1        NaN         audi  gas        std  four        wagon   
8          1        158         audi  gas      turbo  four        sedan   
9          0        NaN         audi  gas      turbo   two    hatchback   

  wheels engine   base  ...  engine-size  system  bore  stroke compression  \
0    rwd  front   88.6  ...          130    mpfi  3.47    2.68         9.0   
1    rwd  front   88.6  ...          130    mpfi  3.47    2.68         9.0   
2    rwd  front   94.5  ...          152    mpfi  2.68    3.47         9.0   
3    fwd  front   99.8  ...          109    mpfi  3.19    3.40        10.0   
4    4wd  front   99.4  ...          136    mpfi  3.19    3.40         8.0   
5    fwd  front   99.8  ...          136    mpfi  3.19    3.40         8.5   
6    fwd  front  105.8  ...          136    mpfi  3.19    3.40         8.5   
7    fwd  front  105.8  ...          136    mpfi  3.19    3.40         8.5   
8    fwd  front  105.8  ...          131    mpfi  3.13    3.40         8.3   
9    4wd  front   99.5  ...          131    mpfi  3.13    3.40         7.0   

  horsepower  peak City Higway  price  
0        111  5000   21     27  13495  
1        111  5000   21     27  16500  
2        154  5000   19     26  16500  
3        102  5500   24     30  13950  
4        115  5500   18     22  17450  
5        110  5500   19     25  15250  
6        110  5500   19     25  17710  
7        110  5500   19     25  18920  
8        140  5500   17     20  23875  
9        160  5500   16     22    NaN  

[10 rows x 26 columns]

<h3 Style="color:#DAA520">2.2 Contar los valores faltantes por columna </h3>

In [51]:
valoresPerdidos = dfAutos.isnull()

In [52]:
valoresPerdidos.head(5)

symboling  normalized   make   fuel  aspiration  doors  style  wheels  \
0      False        True  False  False       False  False  False   False   
1      False        True  False  False       False  False  False   False   
2      False        True  False  False       False  False  False   False   
3      False       False  False  False       False  False  False   False   
4      False       False  False  False       False  False  False   False   

   engine   base  ...  engine-size  system   bore  stroke  compression  \
0   False  False  ...        False   False  False   False        False   
1   False  False  ...        False   False  False   False        False   
2   False  False  ...        False   False  False   False        False   
3   False  False  ...        False   False  False   False        False   
4   False  False  ...        False   False  False   False        False   

   horsepower   peak   City  Higway  price  
0       False  False  False   False  False  
1       False  False  False   False  False  
2       False  False  False   False  False  
3       False  False  False   False  False  
4       False  False  False   False  False  

[5 rows x 26 columns]

In [56]:
for column in valoresPerdidos.columns.values.tolist():
    print (column)
    print (valoresPerdidos[column].value_counts())
    print ("")

symboling
False    226
Name: symboling, dtype: int64

normalized
False    180
True      46
Name: normalized, dtype: int64

make
False    226
Name: make, dtype: int64

fuel
False    226
Name: fuel, dtype: int64

aspiration
False    226
Name: aspiration, dtype: int64

doors
False    224
True       2
Name: doors, dtype: int64

style
False    226
Name: style, dtype: int64

wheels
False    226
Name: wheels, dtype: int64

engine
False    226
Name: engine, dtype: int64

base
False    226
Name: base, dtype: int64

length
False    226
Name: length, dtype: int64

width
False    226
Name: width, dtype: int64

heingt
False    226
Name: heingt, dtype: int64

curb
False    226
Name: curb, dtype: int64

engineType
False    226
Name: engineType, dtype: int64

Cylinders
False    226
Name: Cylinders, dtype: int64

engine-size
False    226
Name: engine-size, dtype: int64

system
False    226
Name: system, dtype: int64

bore
False    222
True       4
Name: bore, dtype: int64

stroke
False    222
True     

<h3 Style="color:#DAA520">2.3 Imputar datos numericos  </h3> #SE AGREGA IMAGEN O PANTALLAZO QUE NECESITAMOS PARA EL ANALISIS CON EL CODIGO DE LA IMAGEN QUE GUARDE EN LA CARPETA DE NUEVO 
<img SRC="EXTRACTO.PNG"/>

In [61]:
avg_normalized = dfAutos["normalized"].astype("float").mean(axis=0)
print("la media del normalized:", avg_normalized)

la media del normalized: 122.28333333333333


In [65]:
dfAutos["normalized"].replace(np.nan, 122, inplace=True)
dfAutos.head(3)

symboling normalized         make fuel aspiration doors        style  \
0          3        122  alfa-romero  gas        std   two  convertible   
1          3        122  alfa-romero  gas        std   two  convertible   
2          1        122  alfa-romero  gas        std   two    hatchback   

  wheels engine  base  ...  engine-size  system  bore  stroke compression  \
0    rwd  front  88.6  ...          130    mpfi  3.47    2.68         9.0   
1    rwd  front  88.6  ...          130    mpfi  3.47    2.68         9.0   
2    rwd  front  94.5  ...          152    mpfi  2.68    3.47         9.0   

  horsepower  peak City Higway  price  
0        111  5000   21     27  13495  
1        111  5000   21     27  16500  
2        154  5000   19     26  16500  

[3 rows x 26 columns]

<h2 Style="color:#8B008B">3.Estandarizacion de datos </h2>
<p> Que se hace con esta etapa </p>

<ol> 
    <li>Verificar el dominio de las variables y se ajusta, si es necesario</li>
    <li>Convertir valores en funcion de la interpretacion</li>
    <li>Normalizar los datos para que sus valores queden entre 0 y 1</li>
    <li>Agrupar datos (Binning)</li>
    
</ol>

<h3 Style="color:#8B008B">3.1 Verificar el dominio de la variable </h3>

    